✨ Завершение задания спринта 4: доработки и улучшения MovieQuiz

YuraGvilia opened the pull request:
•	Добавлен сплэш-экран для улучшения пользовательского опыта
	•	Реализована обработка ошибок
	•	Добавлена статистика для отслеживания прогресса игроков
	•	Внесены изменения в интерфейс для повышения удобства использования

Commit:
Реализованы все требования спринта 4: добавлен сплэш-экран, статистика и обработка ошибок

So1ovyov commented on MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -1,3 +1 @@
 import UIKit
```
Кажется, что пустой файл никак не используется и ничего не содержит, давай его удалим из проекта.

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,188 @@
+// MARK: - Структуры моделей данных
+
+/// Модель вопроса квиза
+struct QuizQuestion {
```
Структуры также можно пометить модификатором `private`:
```
    private struct QuizQuestion {
        let image: String
        let text: String
        let correctAnswer: Bool
    }

    private struct QuizStepViewModel {
        let image: UIImage
        let question: String
        let questionNumber: String
    }

    private struct QuizResultsViewModel {
        let title: String
        let text: String
        let buttonText: String
    }
```

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,188 @@
 final class MovieQuizViewController: UIViewController {
+    
+    // MARK: - UI Элементы
+    @IBOutlet weak var counterLabel: UILabel!
```
Давай также инкапсулируем эти свойства в классе, по ТЗ необходимо проставить модификатор `private` всем свойствам и методам в классе.

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,188 @@
+        QuizQuestion(image: "Vivarium", text: "Рейтинг этого фильма\nбольше чем 6?", correctAnswer: false)
+    ]
+
+    private var currentQuestionIndex = 0
```
В языке `swift` много синтаксического сахара, например можно использовать конструкцию `.zero`
```
private var currentQuestionIndex: Int = .zero
private var correctAnswers: Int = .zero
private var totalGamesPlayed: Int = .zero
private var totalCorrectAnswers: Int = .zero
private var bestScore: Int = .zero
```
Это опцониально, в случае если тебе понравится такой стиль.

So1ovyov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,188 @@
+    // MARK: - Логика игры
+    private func showNextQuestion() {
+        guard !questions.isEmpty else {
+            fatalError("Ошибка: Вопросы отсутствуют в массиве!")
```
Использование подобных выражений не приветствуется внутри основного кода приложения, лучше только в тестах:
https://stevenpcurtis.medium.com/precondition-assert-fatal-error-or-guard-in-your-swift-code-5f9297658be0

В твоем случае достаточно `return`
```
guard !questions.isEmpty else { return }
```

Commit:
Исправлены замечания ревьюера:
- Установлен правильный шрифт и отступы в счетчике вопросов
- Добавлены корректные цвета рамки при ответе
- Рамка теперь исчезает при смене вопроса
- Все IBOutlet свойства теперь private
- Добавлен UIColor+Extensions.swift для работы с HEX цветами

YuraGvilia commented:
✅ Исправления после ревью

⚠️ Критические замечания (исправлено!):

1️⃣ Шрифт заголовка вопроса
	•	Было: Использовался неверный шрифт.
	•	Стало: Теперь используется YS Display Bold, размер 23, межбуквенное расстояние 0.35px, выравнивание по центру.

2️⃣ Исправлен отступ у счетчика вопросов (1/10)
	•	Было: Смещен влево.
	•	Стало: Теперь counterLabel выровнен правильно, отступы соответствуют макету.

3️⃣ Скругления у постера и рамки
	•	Было: У изображения постера и рамки отсутствовало скругление.
	•	Стало: Теперь у imageView радиус скругление 20 (соответствует макету).

4️⃣ Использованы корректные цвета рамки при ответах
	•	Было: Рамка окрашивалась в стандартные red / green, которые не соответствовали макету.
	•	Стало: Теперь используются HEX-значения:
	•	✅ Верный ответ: #60C28E (бледно-зелёный).
	•	❌ Неверный ответ: #F56B6C (бледно-красный).

5️⃣ После выбора ответа рамка исчезает перед следующим вопросом
	•	Было: Рамка оставалась на экране.
	•	Стало: Теперь перед сменой вопроса вызывается imageView.layer.borderColor = UIColor.clear.cgColor, что делает рамку невидимой.

6️⃣ Все @IBOutlet теперь private
	•	Было: Модификатор доступа отсутствовал.
	•	Стало: Теперь private, как требует ТЗ.

7️⃣ Предотвращение “спама” кнопками
	•	Было: Можно было быстро нажимать да/нет и “сломать” игру.
	•	Стало: Теперь кнопки isUserInteractionEnabled = false после ответа и снова true после смены вопроса.

🍏 Рекомендации (учтены):
✅ Все структуры (QuizQuestion, QuizResultsViewModel, QuizStepViewModel) теперь private.
✅ Удалён пустой файл UIColor+Extensions.swift, который не использовался.
✅ Заменены Int = 0 на Int = .zero для читаемости кода.
✅ Удалил fatalError(), вместо него теперь return.

So1ovyov commented:
> ✅ Исправления после ревью
> 
> ⚠️ Критические замечания (исправлено!):
> 
> 1️⃣ Шрифт заголовка вопроса • Было: Использовался неверный шрифт. • Стало: Теперь используется YS Display Bold, размер 23, межбуквенное расстояние 0.35px, выравнивание по центру.
> 
> 2️⃣ Исправлен отступ у счетчика вопросов (1/10) • Было: Смещен влево. • Стало: Теперь counterLabel выровнен правильно, отступы соответствуют макету.
> 
> 3️⃣ Скругления у постера и рамки • Было: У изображения постера и рамки отсутствовало скругление. • Стало: Теперь у imageView радиус скругление 20 (соответствует макету).
> 
> 4️⃣ Использованы корректные цвета рамки при ответах • Было: Рамка окрашивалась в стандартные red / green, которые не соответствовали макету. • Стало: Теперь используются HEX-значения: • ✅ Верный ответ: #60C28E (бледно-зелёный). • ❌ Неверный ответ: #F56B6C (бледно-красный).
> 
> 5️⃣ После выбора ответа рамка исчезает перед следующим вопросом • Было: Рамка оставалась на экране. • Стало: Теперь перед сменой вопроса вызывается imageView.layer.borderColor = UIColor.clear.cgColor, что делает рамку невидимой.
> 
> 6️⃣ Все @IBOutlet теперь private • Было: Модификатор доступа отсутствовал. • Стало: Теперь private, как требует ТЗ.
> 
> 7️⃣ Предотвращение “спама” кнопками • Было: Можно было быстро нажимать да/нет и “сломать” игру. • Стало: Теперь кнопки isUserInteractionEnabled = false после ответа и снова true после смены вопроса.
> 
> 🍏 Рекомендации (учтены): ✅ Все структуры (QuizQuestion, QuizResultsViewModel, QuizStepViewModel) теперь private. ✅ Удалён пустой файл UIColor+Extensions.swift, который не использовался. ✅ Заменены Int = 0 на Int = .zero для читаемости кода. ✅ Удалил fatalError(), вместо него теперь return.

Очень крутой формат, спасибо тебе за него, удобно видеть как студент сделал работу над ошибками и ничего не забыл. Молодец!

So1ovyov commented:
А подход к логированию просто шикарный, в будущих спринтах тебе это очень пригодится.

![CleanShot 2025-02-16 at 11 26 06@2x](https://github.com/user-attachments/assets/95ae12d9-e085-4e3f-bd2b-707dbec02842)

So1ovyov commented:
@YuraGvilia отличная работа, образцовый проект, ты очень круто потрудился.
Успехов в следующем спринте!
